gcp_variant_transforms/vcf_to_bq_preprocess.py

```diff
@@ -0,0 +1,129 @@
+  fields, the suggested resolutions and eventually malformed records.
+- Resolved headers file: A VCF file that contains the resolved fields
+  definitions.
+
+The files are generated in the "temp_location" if it is provided. Otherwise,
+they will be saved in the ``directory``.
+
+Run locally:
+python -m gcp_variant_transforms.vcf_to_bq_preprocess \
+  --input_pattern <path to VCF file(s)> \
+  --report_all True
+
+Run on Dataflow:
+python -m gcp_variant_transforms.vcf_to_bq_preprocess \
+  --input_pattern <path to VCF file(s)>
+  --report_all True \
+  --project gcp-variant-transforms-test \
+  --job_name preprocess \
+  --staging_location "gs://integration_test_runs/staging" \
+  --temp_location "gs://integration_test_runs/temp" \
+  --runner DataflowRunner \
+  --setup_file ./setup.py
+"""
+
+import logging
+import sys
+
+import apache_beam as beam
+from apache_beam.options import pipeline_options
+
+from gcp_variant_transforms import vcf_to_bq_common
+from gcp_variant_transforms.libs import conflicts_reporter
+from gcp_variant_transforms.options import variant_transform_options
+from gcp_variant_transforms.transforms import merge_headers
+from gcp_variant_transforms.transforms import merge_header_definitions
+
+_COMMAND_LINE_OPTIONS = [
+    variant_transform_options.FilterOptions,
+    variant_transform_options.PreprocessOptions,
+    variant_transform_options.VcfReadOptions
+]
+
+_PREPROCESS_JOB_NAME = 'preprocess-vcf-files'
+
+
+def _add_inferred_headers(pipeline,  # type: beam.Pipeline
+                          known_args,  # type: argparse.Namespace
+                          merged_header  # type: pvalue.PCollection
+                         ):
+  # type: (...) -> (pvalue.PCollection, pvalue.PCollection)
+  inferred_headers = vcf_to_bq_common.get_inferred_headers(pipeline, known_args,
+                                                           merged_header)
+  merged_header = (
+      (inferred_headers, merged_header)
+      | beam.Flatten()
+      | 'MergeHeadersFromVcfAndVariants' >> merge_headers.MergeHeaders(
+          allow_incompatible_records=True))
+  return inferred_headers, merged_header
+
+
+# TODO(yifangchen): Add an integration test for this pipeline.
+def run(argv=None):
+  # type: (List[str]) -> (str, str)
+  """Runs preprocess pipeline."""
+  logging.info('Command: %s', ' '.join(argv or sys.argv))
+  known_args, pipeline_args = vcf_to_bq_common.parse_args(argv,
+                                                          _COMMAND_LINE_OPTIONS)
+  options = pipeline_options.PipelineOptions(pipeline_args)
+  pipeline_mode = vcf_to_bq_common.get_pipeline_mode(known_args)
+  if (pipeline_mode == vcf_to_bq_common.PipelineModes.SMALL and
+      not known_args.report_all):
+    options.view_as(pipeline_options.StandardOptions).runner = 'DirectRunner'
+  google_cloud_options = options.view_as(pipeline_options.GoogleCloudOptions)
+  if google_cloud_options.job_name:
```

Comment:
I don't think you need to change the job_name here. We needed to do this for merging headers as we were actually running two pipelines (even though the user requested one pipeline). I don't think you need to change the `pipeline_mode` above by the same logic.